src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2610,21 +2612,37 @@ private static bool WillResultBeAcceptable(bool isStale, IndexQueryBase<Blittabl
             return false;
         }
 
-        protected virtual unsafe long CalculateIndexEtag(bool isStale, DocumentsOperationContext documentsContext,
-            TransactionOperationContext indexContext)
+        protected virtual unsafe long CalculateIndexEtag(DocumentsOperationContext documentsContext,
+            TransactionOperationContext indexContext, QueryMetadata q, bool isStale)
         {
             var length = MinimumSizeForCalculateIndexEtagLength();
 
             var indexEtagBytes = stackalloc byte[length];
 
             CalculateIndexEtagInternal(indexEtagBytes, isStale, State, documentsContext, indexContext);
 
+            UseAllDocumentsEtag(documentsContext, q, length, indexEtagBytes);
+
             unchecked
             {
                 return (long)Hashing.XXHash64.Calculate(indexEtagBytes, (ulong)length);
             }
         }
 
+        protected static unsafe void UseAllDocumentsEtag(DocumentsOperationContext documentsContext, QueryMetadata q, int length, byte* indexEtagBytes)
+        {
+            if (q?.HasIncludeOrLoad == true)
+            {
+                Debug.Assert(length > sizeof(long) * 3);
+
+                long* buffer = (long*)indexEtagBytes;
+                var numberOfDocuments = documentsContext.DocumentDatabase.DocumentsStorage.GetNumberOfDocuments(documentsContext);
+                buffer[0] = DocumentsStorage.ReadLastDocumentEtag(documentsContext.Transaction.InnerTransaction);
+                buffer[1] = DocumentsStorage.ReadLastTombstoneEtag(documentsContext.Transaction.InnerTransaction);
+                buffer[2] = numberOfDocuments;
```

Comment:
`buffer[2]` is 

https://github.com/ravendb/ravendb/blob/d0c1c05d8e45b6324d44d0889ae9f75369f9c0b7/src/Raven.Server/Documents/Indexes/Index.cs#L2585

https://github.com/ravendb/ravendb/blob/d0c1c05d8e45b6324d44d0889ae9f75369f9c0b7/src/Raven.Server/Documents/Indexes/Index.cs#L2578

It means you won't detect index changes if there is no updates to documents. That's wrong.

BTW why to override the values here instead of appending them at the end of `indexEtagBytes` buffer?

Reply:
I'm setting this here because this is only valid if there is load / include.
The idea is that I'm using these three values to know if there was any change _to any document_, not just the ones covered by the index.

How can there _be_ index changes if the docs haven't been updated?

Reply:
Because indexing might be still in-progress. Let's say that you put a new index into db with load/include and you start querying it. There are no doc changes but indexing batches change
the last indexed etag

Reply:
https://github.com/ravendb/ravendb/pull/7166

Reply:
Thanks, got distracted